### PR TITLE
[feat] adds case details to the CaseDetails component

### PR DESCRIPTION
**Pull request by varortz**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tanabata_tree: Description
Adds case details for the selected case listing card to the cases page.
### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Centers line dividing the `CardColumn` & the` InterestForm` + `Case Details`
- Adds case fields to the case details 
- Implements helper for parsing and formatting some of the case fields
- Fixes width of radio button borders

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

image

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Go to the cases page and compare it to the figma designs

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

**Review by david-qing — APPROVED** [review]
lgtm

File: src/components/CaseDetails/index.tsx [hunk 1]
```diff
@@ -1,31 +1,114 @@
...
+            <FieldRow>
+              {caseFields.slice(4, 8).map(({ label, value }) => (
+                <div key={label}>{renderField(label, value(caseData))}</div>
+              ))}
+            </FieldRow>
+            <FieldRow>
+              {caseFields.slice(8, 12).map(({ label, value }) => (
+                <div key={label}>{renderField(label, value(caseData))}</div>
+              ))}
+            </FieldRow>
```
Comment (jinkang-0) [inline src/components/CaseDetails/index.tsx]: I think we could just dump these all in (instead of slicing) and specify rows in the grid container

File: src/components/CaseDetails/styles.tsx [hunk 2]
```diff
@@ -32,10 +31,34 @@ export const InfoContainer = styled.div`
...
+export const FieldRow = styled.div`
+  column-gap: 1.3rem;
+  justify-content: space-between;
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr;
+`;
+
+export const StrongP = styled(P)`
+  font-weight: 700;
+`;
```
Comment (jinkang-0) [inline src/components/CaseDetails/styles.tsx]: we might want to move this into text styles and call it just `Strong`, just to stay consistent

File: src/utils/helpers.ts [hunk 3]
```diff
@@ -45,3 +54,71 @@ export const isValidDate = (d: string) => {
...
+export const parseRolesNeeded = (
+  attorney: boolean | undefined,
+  interpreter: boolean | undefined,
+  card: boolean,
+) => {
+  if (interpreter && attorney) {
+    if (card) {
+      return 'Interpreter & Attorney';
+    }
+    return 'Attorney, Interpreter';
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: nit: can we consolidate this into a ternary statement?

File: src/utils/helpers.ts [hunk 4]
```diff
@@ -45,3 +54,71 @@ export const isValidDate = (d: string) => {
...
+  }
+  if (interpreter) {
+    return 'Interpreter';
+  }
+  return 'Attorney';
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: nit: can we consolidate this also into a ternary statement?

File: src/utils/helpers.ts [hunk 5]
```diff
@@ -45,3 +54,71 @@ export const isValidDate = (d: string) => {
...
+ * If the description is used for the case listing cards, CARD should be passed in as
+ * TRUE, FALSE otherwise.
+ * @param attorney - a boolean indicating whether a case needs an attorney.
+ * @param interpreter - a boolean indicating whether a case needs an interpreter.
+ * @param card a boolean indicating whether the description will be used for a case listing card.
+ * @returns appropriate description for roles needed field based on the attorney & interpreter parameters.
+ */
+export const parseRolesNeeded = (
+  attorney: boolean | undefined,
+  interpreter: boolean | undefined,
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: instead of `boolean | undefined`, it may be cleaner to write something like `attorney?: boolean`

File: src/components/ListingCard/index.tsx [hunk 6]
```diff
@@ -54,9 +59,7 @@ export default function ListingCard({
       onClick={onClick ? () => onClick(caseData.id) : undefined}
     >
       <H4>
-        {caseData.title
-          ? caseData.title
-          : 'Guatemalan mother and two children seeking protection'}
+        {caseData.title ? caseData.title : 'Migrant seeking representation'}
```
Comment (jinkang-0) [inline src/components/ListingCard/index.tsx]: nit: we can refactor this to `caseData.title || 'Migrant seeking representation'`

File: src/utils/helpers.ts [hunk 7]
```diff
@@ -1,8 +1,13 @@
 /* eslint-disable no-bitwise */
-// inspiration from https://stackoverflow.com/a/57941711/22063638
-// example timestamp format: 2024-01-18T11:22:40+00:00
-// WARNING: assumes +00:00 (which should be the case for timestamptz)
-export const timestampStringToDate = (ts: string): Date => {
+
+/**
+ *  * inspiration from https://stackoverflow.com/a/57941711/22063638
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: nit: can we remove this extra `*` lol

File: src/utils/helpers.ts [hunk 8]
```diff
@@ -1,8 +1,13 @@
...
+ * example timestamp format: 2024-01-18T11:22:40+00:00
+ * WARNING: assumes +00:00 (which should be the case for timestamptz)
+ * @param ts
+ * @returns
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: returns `Date`

File: src/utils/helpers.ts [hunk 9]
```diff
@@ -45,3 +54,65 @@ export const isValidDate = (d: string) => {
   }
   return false;
 };
+
+/**
+ * @param experience - an attorney's required experience for a listed case.
+ * @returns appropriate string description according to the experience level.
+ */
+export const parseExperience = (experience: string) => {
```
Comment (jinkang-0) [inline src/utils/helpers.ts]: nit: can `experience` be a `ImmigrationLawExperienceEnum`?

File: src/components/CaseDetails/index.tsx [hunk 10]
```diff
@@ -1,31 +1,100 @@
...
+    <P>{label}</P>
+    <StrongP>{value}</StrongP>
+  </FieldContainer>
+);
+
+// CLEANER IMPLEMENTATION
+const caseFields = [
+  {
+    label: 'Needs',
+    value: (data: CaseListing) =>
```
Comment (jinkang-0) [inline src/components/CaseDetails/index.tsx]: nit: should we use a different name other than `value`? `value` sounds like it's just a value as opposed to a function

**Review by jinkang-0 — APPROVED** [review]
